src/Data/Array/Mutable.idr

```diff
@@ -283,22 +283,37 @@ parameters {n : Nat}
 --------------------------------------------------------------------------------
 
 parameters {n : Nat}
-           (f : F1 s a -> F1 s b)
+           (f : a -> F1 s b)
            (r : MArray s n a)
 
   ||| Apply a function `f` to each element of the mutable array.
   export
-  mmap : F1 s (MArray s n b)
-  mmap t =
+  mmap1 : F1 s (MArray s n b)
+  mmap1 t =
     let tmt # t := unsafeMArray1 n t
-        _   # t := genFrom' tmt n (f . go) t
+        _   # t := genFrom' tmt n go t
       in tmt # t
     where
       go :  Fin n
-         -> F1 s a
+         -> F1 s b
+      go x t =
+        let x'  # t := get r x t
+            x'' # t := f x' t
+          in x'' # t
+
+parameters {n : Nat}
+           (f : a -> b)
+           (r : MArray s n a)
```

Comment:
I suggest to only use `parameters` block if you are actually going to reuse the parameters in several functions. Otherwise, it makes more sense to just use a single function signature in my opinion.

Reply:
@stefan-hoeck 

Nice catch here, this is addressed.